client/packages/invoices/src/Prescriptions/DetailView/Footer/StatusChangeButton.tsx

```diff
@@ -129,44 +143,56 @@ const useStatusChangeButton = () => {
     selectedOption,
     setSelectedOption,
     getConfirmation,
-    lines,
+    isEmptyLines,
     isDisabled,
+    showPaymentWindow,
+    onConfirmStatusChange,
   };
 };
 
 export const StatusChangeButton = () => {
+  const t = useTranslation();
+  const { onOpen, onClose, isOpen } = useEditModal<Draft>();
+
   const {
     options,
     selectedOption,
     setSelectedOption,
     getConfirmation,
-    lines,
+    isEmptyLines,
     isDisabled,
+    showPaymentWindow,
+    onConfirmStatusChange,
   } = useStatusChangeButton();
-  const t = useTranslation();
-  const noLines =
-    lines?.totalCount === 0 ||
-    lines?.nodes?.every(l => l.type === InvoiceLineNodeType.UnallocatedStock);
 
-  const noLinesNotification = useDisabledNotificationToast(
```

Comment:
Moved `noLines` within the `useStatusChangeButton` hook as majority of business logic relating to a line is managed there